compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -418,9 +420,40 @@ public void visit(BLangRecordTypeNode recordTypeNode) {
             rewrite(field, recordTypeNode.typeDefEnv);
         }
         recordTypeNode.restFieldType = rewrite(recordTypeNode.restFieldType, env);
+        generateClosuresForDefaultValuesInTypeRefs(recordTypeNode, typeSymbol);
         result = recordTypeNode;
     }
 
+    private void generateClosuresForDefaultValuesInTypeRefs(BLangRecordTypeNode recordTypeNode,
+                                                            BTypeSymbol typeSymbol) {
+        Map<String, BInvokableSymbol> defaultValues = ((BRecordTypeSymbol) typeSymbol).defaultValues;
+        String typeName = recordTypeNode.symbol.name.value;
+        for (BLangType type : recordTypeNode.typeRefs) {
+            BRecordType recordType = (BRecordType) Types.getReferredType(type.getBType());
+            Map<String, BInvokableSymbol> defaultValuesOfTypeRef =
+                                                                ((BRecordTypeSymbol) recordType.tsymbol).defaultValues;
+            for (Map.Entry<String, BInvokableSymbol> defaultValue : defaultValuesOfTypeRef.entrySet()) {
+                String name = defaultValue.getKey();
+                if (defaultValues.containsKey(name)) {
+                    continue;
+                }
+                BInvokableSymbol symbol = defaultValue.getValue();
+                BLangInvocation invocation = getFunctionPointerInvocation(symbol);
+                String closureName = RECORD_DELIMITER + typeName + RECORD_DELIMITER + name;
+                generateClosureForDefaultValues(closureName, name, invocation, symbol.retType, typeSymbol);
+            }
+        }
+    }
+
+    private BLangInvocation getFunctionPointerInvocation(BInvokableSymbol symbol) {
```

Comment:
Isn't this method a duplicate of `org.wso2.ballerinalang.compiler.desugar.ClosureGenerator#getInvocation`?